cupyx/scipy/ndimage/filters.py

```diff
@@ -373,10 +313,21 @@ def _get_nd_kernel(name, pre, found, post, mode, wshape, int_type,
     cond = ' || '.join(['(ix_{0} < 0)'.format(j) for j in range(ndim)])
     expr = ' + '.join(['ix_{0}'.format(j) for j in range(ndim)])
 
+    if has_weights:
+        weights_init = 'const W* weights = (const W*)&w[0];\nint iw = 0;'
+        weights_check = 'W wval = weights[iw++];\nif (wval)'
+    else:
+        in_params = 'raw X x'
+        weights_init = weights_check = ''
+
     loops = []
     for j in range(ndim):
-        boundary = _generate_boundary_condition_ops(mode, 'ix_{}'.format(j),
-                                                    'xsize_{}'.format(j))
+        if wshape[j] == 1:
+            loops.append('{{ {type} ix_{j} = ind_{j} * xstride_{j};'.
+                         format(j=j, type=int_type))
+        else:
+            boundary = _generate_boundary_condition_ops(
+                mode, 'ix_{}'.format(j), 'xsize_{}'.format(j))
         loops.append('''
```

Comment:
```suggestion
            loops.append('''
```

Reply:
Thanks for refactoring to only avoid duplicate 1d and nd code paths!

I can confirm good performance of the refactored kernel. The only problem I encountered was the indentation error above.
The 1Dd case now performs approximately the same as what I had in `cupyimg`. 3D cases I tested are up to 25% faster for this kernel.

Reply:
please address this!
thanks 🙂

Reply:
Completed in #3390